core/connection/manager.go

```diff
@@ -484,12 +492,17 @@ func (manager *connectionManager) consumeConnectionStates(stateChannel <-chan St
 	logDisconnectError(manager.Disconnect())
 }
 
-func (manager *connectionManager) consumeStats(statisticsChannel <-chan consumer.SessionStatistics) {
-	for stats := range statisticsChannel {
-		manager.eventPublisher.Publish(StatisticsEventTopic, SessionStatsEvent{
-			Stats:       stats,
-			SessionInfo: manager.getCurrentSession(),
-		})
+func (manager *connectionManager) consumeStats(statisticsChannel <-chan consumer.SessionStatistics, stop <-chan struct{}) {
```

Comment:
could we perhaps return an "unsubscribe" method instead of passing in a stop channel?

Passing in another channel only adds to confusion

Reply:
Did it with a returned unsubscribe function.